lib/npg_pipeline/function/bqsr_calc.pm

```diff
@@ -0,0 +1,170 @@
+package npg_pipeline::function::bqsr_calc;
+
+use namespace::autoclean;
+
+use Moose;
+use MooseX::StrictConstructor;
+use Readonly;
+
+use npg_pipeline::cache::reference;
+use npg_pipeline::function::definition;
+
+extends 'npg_pipeline::base';
+with qw{ npg_pipeline::function::util
+         npg_pipeline::product::release };
+
+Readonly::Scalar my $FUNCTION_NAME => 'bqsr_calc';
+
+Readonly::Scalar my $GATK_EXECUTABLE => 'gatk';
+Readonly::Scalar my $GATK_TOOL_NAME => 'BaseRecalibrator';
+
+Readonly::Scalar my $FS_NUM_SLOTS                 => 2;
+Readonly::Scalar my $MEMORY                       => q{2000}; # memory in megabytes
+Readonly::Scalar my $CPUS                         => 1;
+Readonly::Scalar my $NUM_HOSTS                    => 1;
+
+
+our $VERSION = '0';
+
+=head2 create
+
+  Arg [1]    : None
+
+  Example    : my $defs = $obj->create
+  Description: Create per-product data file function definitions.
+
+  Returntype : ArrayRef[npg_pipeline::function::definition]
+
+=cut
+
+sub create {
+  my ($self) = @_;
+
+  my $id_run = $self->id_run();
+  my $job_name = sprintf q{%s_%d}, $FUNCTION_NAME, $id_run;
+
+  # If BQSR is disabled we should not run.
+  my @products = grep { $self->bqsr_enable($_) }
+                 @{$self->products->{data_products}};
+
+  if (scalar @products == 0) {
+    $self->debug('no BQSR enabled data products, skipping');
+  }
+  my @definitions = ();
+
+
+  foreach my $product (@products) {
+    $self->is_release_data($product) or next;
```

Comment:
Might be better to move this filter as an extra grep in lines 47-48.

Reply:
done